### PR TITLE
API-8684: Add info about accesible threads

### PR DESCRIPTION
**Pull request by Adimus11**
## 🚀 Links

- [Feature branch](https://developers.labs.livechat.com/docs/feature/API-8684-add-info-about-accesible-threads)
- [Jira](https://livechatinc.atlassian.net/browse/XXX-XXX)

## ✅ Checklist (for API docs)

- Changelog
- API versions
- Web & RTM
- **Table:** Available methods/webhooks/pushes
- **Table:** Specifics

For more guidelines, see [Updating API docs](https://livechatinc.atlassian.net/wiki/spaces/PAT/pages/761529400/Updating+API+docs).

### Content

- New content
- Proofreading/content fixes
  
### Features (for the website)

- New feature
- Bug fix
- Breaking change

### CODE REVIEW

**Review by lpmichalik — APPROVED** [review]

File: content/messaging/agent-chat-api/v3.4/index.mdx [hunk 1]
```diff
@@ -501,17 +501,19 @@ Properties are **key-value storages**. They can be set within a chat, a thread,
...
+| `events`                        | optional  | Doesn't exists if `restricted_access` is present.                                                                  |
+| `properties`                    | optional  |                                                                                                                    |
+| `restricted_access`             | optional  | Contains the reason for access restriction.                                                                        |
+| `queue`                         | optional  | Present only if the chat is in the queue. The wait time for an available agent is approximated in seconds.         |
+| `previous_thread_id`            | optional  | Present only if there is a preceding thread.                                                                       |
+| `next_thread_id`                | optional  | Present only if there is a following thread.                                                                       |
+| `previous_accessible_thread_id` | optional  | Present only if there is a preceding thread accessible for user. Only present in [`list_archives`](#list-archives) |
```
Comment (oliwiapolec) [inline content/messaging/agent-chat-api/v3.4/index.mdx:514]:
```suggestion
| `previous_accessible_thread_id` | optional  | Present only if there is a preceding thread accessible for the user. Field present only in [`list_archives`](#list-archives) |
```

File: content/messaging/agent-chat-api/v3.4/index.mdx [hunk 2]
```diff
@@ -501,17 +501,19 @@ Properties are **key-value storages**. They can be set within a chat, a thread,
...
+| `properties`                    | optional  |                                                                                                                    |
+| `restricted_access`             | optional  | Contains the reason for access restriction.                                                                        |
+| `queue`                         | optional  | Present only if the chat is in the queue. The wait time for an available agent is approximated in seconds.         |
+| `previous_thread_id`            | optional  | Present only if there is a preceding thread.                                                                       |
+| `next_thread_id`                | optional  | Present only if there is a following thread.                                                                       |
+| `previous_accessible_thread_id` | optional  | Present only if there is a preceding thread accessible for the requester. Field present only in [`list_archives`](#list-archives) |
+| `next_accessible_thread_id`     | optional  | Present only if there is a following thread accessible for user. Only present in [`list_archives`](#list-archives) |
```
Comment (oliwiapolec) [inline content/messaging/agent-chat-api/v3.4/index.mdx:515]:
```suggestion
| `next_accessible_thread_id`     | optional  | Present only if there is a following thread accessible for the requester. Field present only in [`list_archives`](#list-archives) |
```

**Review by oliwiapolec — APPROVED** [review]

**Review by ale-kacperczyk — COMMENTED** [review]
3 suggestions:
1. I believe the snippet with `Thread` should also be updated.
2. What about RTM?
3. Shouldn't we mention these new fields in Changelog?

**Comment by Adimus11** [comment]
@ale-kacperczyk Some improvements implemented based on this 3 ideas

File: content/messaging/agent-chat-api/v3.4/index.mdx [hunk 3]
```diff
@@ -501,17 +501,19 @@ Properties are **key-value storages**. They can be set within a chat, a thread,
...
+| `properties`                    | optional  |                                                                                                                    |
+| `restricted_access`             | optional  | Contains the reason for access restriction.                                                                        |
+| `queue`                         | optional  | Present only if the chat is in the queue. The wait time for an available agent is approximated in seconds.         |
+| `previous_thread_id`            | optional  | Present only if there is a preceding thread.                                                                       |
+| `next_thread_id`                | optional  | Present only if there is a following thread.                                                                       |
+| `previous_accessible_thread_id` | optional  | Present only if there is a preceding thread accessible for the requester. Field present only in [`list_archives`](#list-archives) |
+| `next_accessible_thread_id`     | optional  | Present only if there is a following thread accessible for the requester. Field present only in [`list_archives`](#list-archives) |
```
Comment (oliwiapolec) [inline content/messaging/agent-chat-api/v3.4/index.mdx:515]:
```suggestion
| `next_accessible_thread_id`     | optional  | Present only if there is a following thread accessible for the requester. Field present only in [`list_archives`](#list-archives). |
```

File: content/messaging/agent-chat-api/v3.4/rtm-reference/index.mdx [hunk 4]
```diff
@@ -538,18 +538,20 @@ Properties are **key-value storages**. They can be set within a chat, a thread,
...
+| `restricted_access`  | optional  | Contains the reason for access restriction.                                                                                                  |
+| `queue`              | optional  | Present only if the chat is in the queue. The wait time for an available agent is approximated in seconds.                                   |
+| `queues_duration`    | optional  | Present only in the [List Archives](#list-archives) responses if the chat was queued.                                                        |
+| `previous_thread_id` | optional  | Present only if there is a preceding thread.                                                                                                 |
+| `next_thread_id`     | optional  | Present only if there is a following thread.                                                                                                 |
+| `previous_accessible_thread_id` | optional  | Present only if there is a preceding thread accessible for the requester. Field present only in [`list_archives`](#list-archives) |
```
Comment (oliwiapolec) [inline content/messaging/agent-chat-api/v3.4/rtm-reference/index.mdx:552]:
```suggestion
| `previous_accessible_thread_id` | optional  | Present only if there is a preceding thread accessible for the requester. Field present only in [`list_archives`](#list-archives). |
```

File: content/messaging/agent-chat-api/v3.4/rtm-reference/index.mdx [hunk 5]
```diff
@@ -538,18 +538,20 @@ Properties are **key-value storages**. They can be set within a chat, a thread,
...
+| `next_accessible_thread_id`     | optional  | Present only if there is a following thread accessible for the requester. Field present only in [`list_archives`](#list-archives) |
```
Comment (oliwiapolec) [inline content/messaging/agent-chat-api/v3.4/rtm-reference/index.mdx:553]:
```suggestion
| `next_accessible_thread_id`     | optional  | Present only if there is a following thread accessible for the requester. Field present only in [`list_archives`](#list-archives). |
```

**Comment by oliwiapolec** [comment]
@Adimus11 https://developers.labs.livechat.com/docs/feature/API-8684-add-info-about-accesible-threads/messaging/agent-chat-api/v3.4/#threads will this snippet be updated? 👀

**Comment by Adimus11** [comment]
@oliwiapolec I don't think so, as `*_thread_id` properties are connected strictly with archives, so representing them in general structure seems a little bit wrong, WDYT?

**Comment by oliwiapolec** [comment]
@Adimus11 cool, that's reasonable for me. As long as it's added to the list_archives response I'm good with that :D

File: content/messaging/agent-chat-api/changelog/index.mdx [hunk 6]
```diff
@@ -24,6 +24,10 @@ The developer preview version provides a preview of the upcoming changes to the
   - [**thread_deleted**](/messaging/agent-chat-api/v3.4/rtm-reference/#thread_deleted)
   - [**threads_deleted**](/messaging/agent-chat-api/v3.4/rtm-reference/#threads_deleted)
 
+### Threads
+
+- There are new properties: `previous_accessible_thread_id` and `next_accessible_thread_id`, which can be returned by the **List Archives** method, ([Web](/messaging/agent-chat-api/v3.4/#list-archives) and [RTM](/messaging/agent-chat-api/v3.4/rtm-reference/#list-archives))
```
Comment (oliwiapolec) [inline content/messaging/agent-chat-api/changelog/index.mdx:29]:
```suggestion
- There are new properties: `previous_accessible_thread_id` and `next_accessible_thread_id`, which can be returned by the **List Archives** method, ([Web](/messaging/agent-chat-api/v3.4/#list-archives) and [RTM](/messaging/agent-chat-api/v3.4/rtm-reference/#list-archives)).
```

**Review by ale-kacperczyk — APPROVED** [review]

**Pull request merged**